Cloudflare/canary-receiver.js

```diff
@@ -71,12 +71,29 @@ async function handleRequest(request) {
       }
 
       if (data.CanaryName == MyCanary) {
-        const MaliciousIP = data.SourceIP
-        // process  
-        // Store in KV store (Key-Value store)
-        await canaryblocks.put(MaliciousIP, Timestamp, {expirationTtl: IPBlocklistTTL})
+        // process
+        switch (true){
+          case /Scan/.test(data.Description):
+            // Store in KV store (Key-Value store)
+            await canaryblocks.put(MaliciousIP, Timestamp, {expirationTtl: IPBlocklistTTL})
+          break;
+          case /Load/.test(data.Description):
+            break;
+          case /Settings Changed/.test(data.Description):
+            break;
+          case /Disconnected/.test(data.Description):
+            break;
+          case /Reconnected/.test(data.Description):
+            break;
+          case /Dummy/.test(data.Description):
+            break;
+          default:
+            // Store in KV store (Key-Value store)
+            await canaryblocks.put(MaliciousIP, Timestamp, {expirationTtl: IPBlocklistTTL})
+          break;
+        } // end switch
```

Comment:
If i'm following here you'd like to exclude these IPs from the block list?

I think that's reasonable. Just a few things:

- /Load/ is going to match against legit alerts, eg. our HTTP Page Load incident will get dropped here. So we'll have to remove that. The others look fine though.
- /Scan/ just looks like the default case in the switch so i think we can clean this up a little bit.
- I think we can do this without a switch statement. It's a little difficult to follow the flow here.
- We're breaking out of the switch and then replying with the ``Marked IP: ${MaliciousIP} and Stored Event: ${Payload}`,` response which isn't happening, right?

Let's rework this slightly to into an if statement with a function:

```
const isHouseKeepingAlert = () => {
  return Settings Changed/.test(data.Description) || /Disconnected/.test(data.Description) || /Reconnected/.test(data.Description) || /Dummy/.test(data.Description);
};

if (isHouseKeepingAlert()) {
  // return response here with the message something like... 'Not marking house keeping alert but storing event for syslog'
}

await canaryblocks.put(MaliciousIP, Timestamp, {expirationTtl: IPBlocklistTTL})

etc...
```
Let me know what you think?

Reply:
Good timing, and I was making some adjustments to our deployed workers and removed the first switch case, since it was redundant.  I like your version much better.  Thank you.